ecms_base/modules/custom/ecms_api/src/EcmsApiBase.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\ecms_api;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Url;
+use Drupal\jsonapi_extras\EntityToJsonApi;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\GuzzleException;
+
+/**
+ * Class EcmsApiBase.
+ *
+ * The EcmsApiBase class is an abstract that can be extended by other
+ * services. By default, this class provides the functionality to authenticate
+ * to another site with Simple OAuth using the
+ * `client_credentials` connection method.
+ * This class also provides the method to submit an entity to another site
+ * using the Json API.
+ *
+ * @package Drupal\ecms_api
+ */
+abstract class EcmsApiBase {
+
+  /**
+   * The API endpoint prefix.
+   */
+  const API_ENDPOINT = 'EcmsApi';
+
+  /**
+   * Allowed HTTP methods to accept for submission to the API endpoints.
+   */
+  const ALLOWED_HTTP_METHODS = [
+    'POST',
+    'PATCH',
+  ];
+
+  /**
+   * Fields that should not be submitted.
+   */
+  const NO_API_FIELD_NAMES = [
+    'drupal_internal__nid',
+    'drupal_internal__vid',
+    'revision_timestamp',
+    'status',
+    'created',
+    'changed',
+    'promote',
+    'sticky',
+    'path',
+  ];
+
+  /**
+   * The http_client service.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * The jsonapi_extras.entity.to_jsonapi service.
+   *
+   * @var \Drupal\jsonapi_extras\EntityToJsonApi
+   */
+  protected $entityToJsonApi;
+
+  /**
+   * EcmsApiBase constructor.
+   *
+   * @param \GuzzleHttp\ClientInterface $httpClient
+   *   The http_client service.
+   * @param \Drupal\jsonapi_extras\EntityToJsonApi $entityToJsonApi
+   *   The jsonapi_extras.entity.to_jsonapi service.
+   */
+  public function __construct(ClientInterface $httpClient, EntityToJsonApi $entityToJsonApi) {
+    $this->httpClient = $httpClient;
+    $this->entityToJsonApi = $entityToJsonApi;
+  }
+
+  /**
+   * Get an access token to authorize the request.
+   *
```

Comment:
Argument comments should be added here

Reply:
Good catch. Not sure how phpcs missed that?